src/test/java/org/openmrs/contrib/qaframework/automation/AttachmentSteps.java

```diff
@@ -0,0 +1,117 @@
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+import org.apache.commons.lang3.StringUtils;
+import org.openmrs.contrib.qaframework.page.AttachmentsPage;
+import org.openmrs.reference.page.HomePage;
+
+import io.cucumber.java.After;
+import io.cucumber.java.en.And;
+import io.cucumber.java.en.Given;
+import io.cucumber.java.en.Then;
+import io.cucumber.java.en.When;
+
+public class AttachmentSteps extends Steps {
+	
+	private AttachmentsPage attachmentsPage;
+	
+	@After("@selenium")
+	public void destroy() {
+		quit();
+	}
+	
+	@Given("a user logs in, searches a patient and visits Attachment page")
+	public void loadPatientDashboard() {
+		goToLoginPage().login(testProperties.getUsername(), testProperties.getPassword(), testProperties.getLocation());
+		homePage = new HomePage(loginPage);
+		findPatientPage = homePage.goToFindPatientRecord();
+		findPatientPage.clickOnFirstPatient();
+		dashboardPage = findPatientPage.clickOnFirstPatient();
+	}
+	
+	@And("a user clicks on Attachment link")
+	public void userClicksOnAttachmentLink() {
+		attachmentsPage = new AttachmentsPage(dashboardPage);
+		attachmentsPage.clickOnAttachmentLink();
+	}
+	
+	@Then("the system loads Attachment page")
+	public void systemLoadsAttachmentPage() {
+		assertEquals(getElement(patientHeaderId).getText(), getElement(patientHeaderId).getText());
+		assertTrue(textExists("Attachments"));
+	}
+	
+	@And("a user attaches a file")
+	public void userAttachesFile() {
+		attachmentsPage.clickOrDropFile();
+	}
+	
+	@And("a user enters a caption")
+	public void userEntersCaption() {
+		attachmentsPage.addCaption();
+	}
+	
+	@Then("the system enables upload button")
+	public void systemEnablesUploadButton() {
+		String captionText = attachmentsPage.addCaption();
+		if (StringUtils.isNotBlank(captionText)) {
+			assertTrue(textExists("Upload file"));
+		}
+	}
+	
+	@When("a user clicks the upload button")
+	public void userClicksUploadButton() {
+		attachmentsPage.clickOnUploadFile();
+	}
+	
+	@Then("the system uploads the attached file")
+	public void systemUploadsAttachedFile() {
+		userClickOnUploadButton();
+		assertTrue(textExists("Upload file"));
+	}
+	
+	@And("a user attaches a file or enters a caption")
+	public void userAttachesFileOrEntersCaption() {
+		userAttachesFile();
+		userEntersCaption();
+	}
+	
+	@And("a user clicks clear forms button")
+	public void userClicksClearFormsButton() {
+		attachmentsPage.clickClearForms();
+	}
+	
+	@Then("the system clears the form")
+	public void systemClearsForm() {
+		assertTrue(textExists("Clear forms"));
+	}
+	
+	@And("a user clicks on the camera icon")
+	public void userClicksOnCameraIcon() {
+		attachmentsPage.clickOnCameraIcon();
+	}
+	
+	@Then("the system enables capturing an image")
+	public void systemEnablesCapturingImage() {
+		attachmentsPage.clickOnCameraIcon();
+		assertNotNull(getElement(AttachmentsPage.ICON_CAMERA));
```

Comment:
this does better coming first

Reply:
Moved the block to run before the action

Reply:
i mean't the opposite, but if the clickon is happening for the same icon_camera element, you don't need to check it's existence since the clickon will fail if it doesn't exisit